main.tf

```diff
@@ -1,3 +1,280 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+locals {
+  # sleep times definition
+  sleep_time_catalog_create   = "60s"
+  sleep_time_operator_create  = "120s"
+  sleep_time_sampleapp_create = "30s"
+
+  # helm chart names
+  ibm_operator_catalog_chart                 = "ibm-operator-catalog"
+  websphere_liberty_operator_chart           = "websphere-liberty-operator"
+  websphere_liberty_operator_group_chart     = "websphere-liberty-operator-group"
+  websphere_liberty_operator_sampleapp_chart = "websphere-liberty-operator-sampleapp"
+
+  # validation of ws_liberty_operator_target_namespace - if null the value of ws_liberty_operator_namespace must be equal to "openshift-operators" https://www.ibm.com/docs/en/was-liberty/core?topic=operator-installing-red-hat-openshift-cli#in-t-cli__install-op-cli__title__1
+  default_liberty_operator_namespace = "openshift-operators"
+  operator_target_namespace_cnd      = var.ws_liberty_operator_target_namespace == null && var.ws_liberty_operator_namespace != local.default_liberty_operator_namespace
+  operator_target_namespace_msg      = "if input var ws_liberty_operator_target_namespace is null the value of ws_liberty_operator_namespace must be equal to ${local.default_liberty_operator_namespace}"
+  # tflint-ignore: terraform_unused_declarations
+  operator_target_namespace_chk = regex("^${local.operator_target_namespace_msg}$", (!local.operator_target_namespace_cnd ? local.operator_target_namespace_msg : ""))
+}
+
+data "ibm_container_cluster_config" "cluster_config" {
+  cluster_name_id = var.cluster_id
+  config_dir      = "${path.module}/kubeconfig"                                                             # See https://github.ibm.com/GoldenEye/issues/issues/552
+  endpoint_type   = var.cluster_config_endpoint_type != "default" ? var.cluster_config_endpoint_type : null # null represents default
+}
+
+# creating the namespace to deploy the helm releases to install the WebSphere Liberty Operator
+resource "kubernetes_namespace" "helm_release_operator_namespace" {
```

Comment:
why do we need this? The helm provider supports passing `create_namespace = true`

Reply:
explained here
https://github.com/terraform-ibm-modules/terraform-ibm-websphere-liberty-operator/pull/3#discussion_r1483226563